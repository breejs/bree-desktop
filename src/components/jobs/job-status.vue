<script setup>
import { computed } from 'vue';

const props = defineProps({
  status: {
    validator(value) {
      return ['running', 'waiting', 'stopped', 'error'].includes(value);
    },
    default: 'error'
  }
});

const STATE_TO_COLOR = new Map([
  ['running', 'success'],
  ['waiting', 'light'],
  ['stopped', 'warning'],
  ['error', 'danger']
]);

const classObject = computed(() => {
  return `text-${STATE_TO_COLOR.get(props.status)}`;
});
</script>

<template lang="pug">
i.bi.bi-circle-fill(:class='classObject')
</template>
