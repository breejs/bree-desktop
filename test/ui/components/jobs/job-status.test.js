import { storyTests } from '../../_story-tests';

import * as stories from '@/components/jobs/job-status.stories';

storyTests(stories);
