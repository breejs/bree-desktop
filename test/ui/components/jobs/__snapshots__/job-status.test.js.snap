// Vitest Snapshot v1

exports[`story snapshots > Default 1`] = `"<i class=\\"bi bi-circle-fill text-danger\\"></i>"`;

exports[`story snapshots > Error 1`] = `"<i class=\\"bi bi-circle-fill text-danger\\"></i>"`;

exports[`story snapshots > Running 1`] = `"<i class=\\"bi bi-circle-fill text-success\\"></i>"`;

exports[`story snapshots > Stopped 1`] = `"<i class=\\"bi bi-circle-fill text-warning\\"></i>"`;

exports[`story snapshots > Waiting 1`] = `"<i class=\\"bi bi-circle-fill text-light\\"></i>"`;
